Tasks/GitHubPRComment/githubComment/index.ts

```diff
@@ -3,68 +3,131 @@ import gitClient = require('@octokit/rest');
         }
 
-        if (path.extname(fileName) == extName){
+        if (path.extname(fileName) == extName) {
             result.push(fileName);
-        } 
+        }
     });
 
     return result;
-}
+};
 
 const combineMessageBody = (files: string[]): string => {
-    var body:string = "";
-    files.forEach(file => {
+    var body: string = "";
+    files.forEach((file) => {
         var bodyFile = fs.readFileSync(file);
-       // var fileObject = JSON.parse(bodyFile.toString());
-       // body += fileObject["body"].toString() + "\r\n";
-       body += bodyFile + "\r\n";
+        // var fileObject = JSON.parse(bodyFile.toString());
+        // body += fileObject["body"].toString() + "\r\n";
+        body += bodyFile + "\r\n";
     });
     return body;
+};
+
+/*
+ *Deletes previous automatically generated comments that contain the comment flag.
+ */
+const deleteComments = async () => {
+    await clientWithAuth.issues
+        .listCommentsForRepo(listCommentParams)
+        .then((res) => {
+            //We don't iterate the whole collection in order to not remove the last comment.
+            for (let index = 0; index < res.data.length - 1; index++) {
+                const element = res.data[index];
+                if (element.body.indexOf(flagComment) != -1) {
+                    deleteCommentParams.comment_id = element.id;
+                    clientWithAuth.issues.deleteComment(deleteCommentParams);
+                }
+            }
+        })
+        .catch((err) => {
+            taskLibrary.setResult(taskLibrary.TaskResult.Failed, err);
+        });
+};
+
+const clientWithAuth = new gitClient({
+    auth: "token " + taskLibrary.getInput("userToken"),
+    userAgent: "octokit/rest.js v1.2.3",
+});
+
+var files = getFilesFromDir(
+    taskLibrary.getInput("bodyFilePath")!, '.' + taskLibrary.getInput('extension'),
+    taskLibrary.getBoolInput("getSubFolders")
+);
+var message = combineMessageBody(files);
+var repo = taskLibrary.getInput('repository')!.split('/');
+var keepCommentHistory = taskLibrary.getBoolInput("keepCommentHistory");
+
+const comment: gitClient.IssuesCreateCommentParams = {
+    owner: repo[0],
+    repo: repo[1],
+    number: parseInt(taskLibrary.getInput('prNumber')!),
+    body: "\r\n" + message + "\r\n" + flagComment,
+};
+
+const listCommentParams: gitClient.IssuesListCommentsForRepoParams = {
+    owner: repo[0],
+    repo: repo[1],
+};
+
+const deleteCommentParams: gitClient.IssuesDeleteCommentParams = {
+    owner: repo[0],
+    repo: repo[1],
+    comment_id: 0,
+};
+
+async function run() {
+
+    if (message) {
+        await clientWithAuth.issues
+            .createComment(comment)
+            .then((res) => {
+                console.log(res);
+            })
+            .catch((err) => {
+                taskLibrary.setResult(taskLibrary.TaskResult.Failed, err);
+            });
+
+        if (!keepCommentHistory) {
+            deleteComments();
+        }
```

Comment:
I think we should move this into the 
```
.createComment(comment)
            .then((res) => {
                console.log(res);
            })
```

because, if it fails to create the comment, it shouldnt delete anything. Its just a suggestion, we can disscuss it here.

Reply:
Yes, I think you're right. Additionally, we can catch a possible exception. Thanks!